workflow/setup_expt.py

```diff
@@ -182,19 +233,20 @@ def edit_baseconfig(host, inputs):
         extend_dict = {
             "@CASEENS@": f'C{inputs.resens}',
             "@NMEM_ENKF@": inputs.nens,
+            "@DOHYBVAR@": "YES" if inputs.nens > 0 else "NO",
         }
         tmpl_dict = dict(tmpl_dict, **extend_dict)
 
+    # Determine CCPP suite and MP based on mode and app
+    gfsv16 = {"@CCPP_SUITE@": "FV3_GFS_v16", "@IMP_PHYSICS@": 11}
+    gfsv17 = {"@CCPP_SUITE@": "FV3_GFS_v17_p8", "@IMP_PHYSICS@": 8}
     if inputs.mode in ['cycled']:
-        extend_dict = {
-            "@CCPP_SUITE@": 'FV3_GFS_v16',
-            "@IMP_PHYSICS@": 11
-        }
+        if inputs.app in ['ATM']:
+            extend_dict = gfsv16
+        elif inputs.app in ['S2S', 'S2SW']:
+            extend_dict = gfsv17
```

Comment:
I thought ATM cycling didn't work with Thompson. How is it working here for S2S apps? If it works, we can probably just move to v17 across the board.

Reply:
It does not work, but I placed a conditional in the `forecast_postdet.sh` See the block following line 52 in `forecast_postdet.sh`